crates/eframe/src/epi.rs

```diff
@@ -672,6 +674,28 @@ impl Frame {
         self.storage.as_deref()
     }
 
+    /// Request the current frame's pixel data. Needs to be retrieved by calling [`eframe::Frame::frame_pixels`]
+    /// during [`eframe::App::post_rendering`].
+    pub fn request_pixels(&mut self) {
+        self.output.pixels_requested = true;
+    }
+
+    /// Cancel a request made with [`eframe::Frame::request_pixels`].
+    pub fn cancel_request_pixels(&mut self) {
+        self.output.pixels_requested = false;
+    }
+
+    /// During [`eframe::App::post_rendering`], use this to retrieve the pixel data that was requested during
+    /// [`eframe::App::update`] via [`eframe::Frame::request_pixels`]. Currently only implemented with wgpu backend.
+    /// Returns None if
+    ///     Called in [`eframe::App::update`]
+    ///     [`eframe::Frame::request_pixels`] wasn't called on this frame during [`eframe::App::update`]
+    ///     The rendering backend doesn't support this feature (yet). Currently only implemented for the wgpu backend.
+    ///     Retrieving the data was unsuccesful in some way.
+    pub fn frame_pixels(&self) -> Option<Vec<u8>> {
```

Comment:
This needs to document the format (premultiplied RGBA), or switch it to using `egui::Color32`.

Does the pixel data start from the top of the screen, or the bottom?

Reply:
Great points! I ended up refactoring to emit an egui::ColorImage from both backends to avoid ambiguity